### ISSUE TITLE
Code review and suggested modifications

### ISSUE THREAD

**Issue by gmuraleekrishna** [meghnatolani/AlienApplication#1]
The one thing that is impressive is the eloquent README given for the project, (excluding the problem statement part). The problem break down is logical and is cleanly done.

The code looks fine, but there are some places you could improve,  which I would like to address.
- Unwanted `try{...} catch {... }` blocks.
  - Use the exception handling block to wrap only the necessary code. Also avoid using generic exceptions
- The second part of the problem is not covered in the solution. Please go through **Part B** of problem statement.
- Indenting the code is an inevitable practice. Please use right formatting for any code you write.
  - It makes your code readable.

**Comment by meghnatolani** [comment]
Thank you sir, for your valuable comments.
I had made some mistakes in my initial code, which I have modified.

For Part B : I have slight doubts regarding this one.
What I have done currently is, in my AlienWritterType.java (Where the format is checked), if a format   other than text and pdf is called, it will check a text file named "compare.text". If that particular format name (foobar eg) exists in that file, then it would check if the class of that format name (foobarWritter) exists. If it does it will be called.
So what basically the user has to do if he has to plug in another format, he has to create a class with that format, and mention the format in "compare.text".
So basically two things happen with this : 1) If someone by mistakely enters a wrong format name, he would be sent to the start of the application. 2) A new format can be plugged in with just creating a new class and mentioning the format name in the text file.

Now, the thing is with my code, it checks if the class is present or not with the name - (format+"Writter"), but I am not able to figure how to call this class, since the "format" is dynamic.

Also, is this is what expected from Part B? Am I heading in the wrong direction?

**Comment by gmuraleekrishna** [comment]
The direction you are heading is correct. Yes, you will have to dynamically instantiate  the **Writer** object from the class (format) name in the compare.text. 

Please use relevant naming for the file and variable names. 

Can you add a _plugged-in_ and _working_ format in your application? The formats currently in compare.txt are examples of an edge case, where a corresponding format class is not present.

Updating README as you progress, is a suggested improvement.

**Comment by meghnatolani** [comment]
Hello Sir,
I have updated the README as well as plugged in a new format class - htmlWritter.java
Kindly have a look at the code.

**Comment by gmuraleekrishna** [comment]
@meghnatolani 
The solution still demands improvements including, but not limited to the following
- **code indentation**
- Generated html file is **not** an html, Check the HTML class.
- Commit messages
  - Please read [Writing good commit messages](https://github.com/erlang/otp/wiki/Writing-good-commit-messages)

Please review the code before submitting. We are looking for a clean and working solution and not necessarily an early submission.

**Comment by gmuraleekrishna** [comment]
@meghnatolani  Can you give us an estimate on when can you submit the changes?

**Comment by meghnatolani** [comment]
Can I get a day more, for final submission?
I apologise for the commit messages, I wasn't aware of how it should be submitted.
And, regarding indentation I had worked on sublime for that. I am re-checking all the files and will make sure this doesn't happen again.

**Comment by gmuraleekrishna** [comment]
@meghnatolani  Ok, sure. Further, please inform us a bit early if you need time. It is to set right expectations.

**Comment by meghnatolani** [comment]
Sure, sir.

**Comment by meghnatolani** [comment]
Sir, you can have a look at the code.
I hope there are no errors now.
Thank you again.
